kpi-tracker/src/main.rs

```diff
@@ -734,24 +898,28 @@ async fn main() -> anyhow::Result<()> {
         .filter_module(module_path!(), args.log_level) // Only log the current module (main).
         .init();
 
-    let mut db = DB {
-        blocks: HashMap::new(),
-        accounts: HashMap::new(),
-        account_transactions: HashMap::new(),
-        contract_modules: HashMap::new(),
-        contract_instances: HashMap::new(),
-        transaction_account_relations: HashSet::new(),
-        transaction_contract_relations: HashSet::new(),
-    };
+    // Since the database connection is managed by the background task we use a
+    // oneshot channel to get the height we should start querying at. First the
+    // background database task is started which then sends the height over this
+    // channel.
+    let (height_sender, height_receiver) = tokio::sync::oneshot::channel();
+    // Create a channel between the task querying the node and the task logging
+    // transactions.
+    let (block_sender, block_receiver) = tokio::sync::mpsc::channel(100);
+
+    tokio::spawn(async {
+        if let Err(e) = run_db_process(block_receiver, height_sender).await {
+            log::error!("Error happened while running DB process: {}", e);
+        }
+    });
```

Comment:
I assume error handling with retries is going to be part of the followup PR, right?

Reply:
Yes, next round will cycle endpoints and retry on errors and all the other good stuff that 's missing when comparing it to the transaction logger 😄